crates/iota-sdk/examples/publish_upgrade.rs

```diff
@@ -0,0 +1,202 @@
+    builder
+        .upgrade(package_id, res("upgrade_ticket"), data)
+        .name("upgrade_receipt");
+
+    // 3. Finalize the upgrade
+    builder
+        .move_call(Address::FRAMEWORK, "package", "commit_upgrade")
+        .arguments((upgrade_cap_arg, res("upgrade_receipt")));
+
+    // Finalize the PTB
+    let tx = builder
+        .gas_budget(50_000_000)
+        .gas(gas)
+        .clone()
+        .finish()
+        .await?;
+
+    // Perform a dry-run to check if everything is fine
+    let res = client.dry_run_tx(&tx, false).await?;
+    if let Some(err) = res.error {
+        bail!("Dry run failed: {err}");
+    }
+    let Some(effects) = res.effects else {
+        bail!("Dry run failed: no effects");
+    };
+    println!("Effects status (dry run): {:?}", effects.status());
+
+    // Sign and execute the transaction (upgrade the package)
+    println!("Upgrading package");
+    let sig = private_key.sign_transaction(&tx)?;
+    let Some(effects) = client.execute_tx(&[sig], &tx).await? else {
+        bail!("Transaction failed: no effects");
+    };
+    println!("Effects status (upgrade): {:?}", effects.status());
+
+    // Wait some time for the indexer to process the tx
+    tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+
+    // Print the new package version (should now be 2)
+    for changed_obj in effects.as_v1().changed_objects.iter() {
+        match changed_obj.output_state {
+            ObjectOut::PackageWrite { version, .. } => {
+                let pkg_id = changed_obj.object_id;
+                println!("PackageId: {pkg_id}");
+                println!("Package version: {version}")
+            }
+            _ => continue,
+        }
+    }
+
+    Ok(())
+}
+
+// Compiled `first_package` example
+const SERIALIZED_FIRST_PACKAGE: &str = r#"{"modules":["oRzrCwYAAAAKAQAIAggUAxw+BFoGBWBBB6EBwQEI4gJACqIDGgy8A5cBDdMEBgAKAQ0BEwEUAAIMAAABCAAAAAgAAQQEAAMDAgAACAABAAAJAgMAABACAwAAEgQDAAAMBQYAAAYHAQAAEQgBAAAFCQoAAQsACwACDg8BAQwCEw8BAQgDDwwNAAoOCgYJBgEHCAQAAQYIAAEDAQYIAQQHCAEDAwcIBAEIAAQDAwUHCAQDCAAFBwgEAgMHCAQBCAIBCAMBBggEAQUBCAECCQAFBkNvbmZpZwVGb3JnZQVTd29yZAlUeENvbnRleHQDVUlEDWNyZWF0ZV9jb25maWcMY3JlYXRlX3N3b3JkAmlkBGluaXQFbWFnaWMJbXlfbW9kdWxlA25ldwluZXdfc3dvcmQGb2JqZWN0D3B1YmxpY190cmFuc2ZlcgZzZW5kZXIIc3RyZW5ndGgOc3dvcmRfdHJhbnNmZXIOc3dvcmRzX2NyZWF0ZWQIdHJhbnNmZXIKdHhfY29udGV4dAV2YWx1ZQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAgMHCAMJAxADAQICBwgDEgMCAgIHCAMVAwAAAAABCQoAEQgGAAAAAAAAAAASAQsALhELOAACAQEAAAEECwAQABQCAgEAAAEECwAQARQCAwEAAAEECwAQAhQCBAEAAAEOCgAQAhQGAQAAAAAAAAAWCwAPAhULAxEICwELAhIAAgUBAAABCAsDEQgLAAsBEgALAjgBAgYBAAABBAsACwE4AgIHAQAAAQULAREICwASAgIAAQACAQEA"],"dependencies":["0x0000000000000000000000000000000000000000000000000000000000000002","0x0000000000000000000000000000000000000000000000000000000000000001"],"digest":[246,127,102,77,186,19,68,12,161,181,56,248,210,0,91,211,245,251,165,152,0,197,250,135,171,37,177,240,133,76,122,124]}"#;
```

Comment:
I would say either move this to a fixture or just compile it using the `iota` binary in the example.

Reply:
Hmm. But this example necessarily involves an external package, so maybe it should be an exception

Reply:
done